covid_graphs/covid_graphs/scripts/rest.py

```diff
@@ -0,0 +1,105 @@
+from pathlib import Path
+from typing import Dict, List
+
+from flask import Flask, abort, jsonify
+
+from covid_graphs.country_graph import CountryGraph
+from covid_graphs.country_report import create_report
+from covid_graphs.predictions import prediction_db
+
+TITLE = "COVID-19 predictions of Boďová and Kollár"
+CountryGraphsByReportName = Dict[str, List[CountryGraph]]
+
+
+class Rest:
+    def __init__(self, data_dir: Path):
+        # vytvor strukturu pre predikcie a krajiny
+        # vytvor strukturu pre data
+        self.available_predictions = Rest._create_available_predictions_by_date()
+        self.country_reports = Rest._create_country_reports(data_dir)
+        predictions, country_reports_active = Rest._create_predictions(self.country_reports)
+        self.predictions = predictions
+        self.country_reports_active = country_reports_active
+
+    @staticmethod
+    def _create_available_predictions_by_date():
+        result = {}
+        for x in prediction_db.get_prediction_events():
+            result[x.name] = {
+                "label": x.date,
+                "countries": [p.country for p in prediction_db.predictions_for_event(x)],
+            }
+        return result
+
+    @staticmethod
+    def _create_country_reports(data_dir: Path):
+        country_reports = {}
+        for country_short_name in prediction_db.get_countries():
+            if (data_dir / f"{country_short_name}.data").is_file():
+                country_report = create_report(
+                    data_dir / f"{country_short_name}.data", country_short_name
+                )
+                country_reports[country_report.short_name] = country_report
+
+        return country_reports
+
+    @staticmethod
+    def _create_predictions(country_reports):
+        predictions = []
+        country_reports_active = {}
+        for country in prediction_db.get_countries():
+            country_predictions = prediction_db.predictions_for_country(country)
+            country_report = country_reports[country]
+            graph = CountryGraph(report=country_report, country_predictions=country_predictions)
+            country_reports_active[country] = {
+                "type": "cumulative_active",
+                "date_list": graph.cropped_dates,
+                "values": graph.cropped_cumulative_active.tolist(),
+                "short_name": country_report.short_name,
+                "long_name": country_report.long_name,
+            }
+            for prediction, curve in zip(country_predictions, graph.curves):
+                xs, ys = curve.generate_trace(graph.display_until)
```

Comment:
Our API is insufficient, you shouldn't be calling `generate_trace` here. We should fix it in https://github.com/lukipuki/COVID-19-simulation/issues/68

Reply:
According to wikipedia definition, I'd say it's more HTTP+JSON than rest, almost because our resources are read-only. However most attributes applies to this service.

Reply:
No, we'll fix this part once #68 is done